sdks/python/apache_beam/io/gcp/bigquery_file_loads.py

```diff
@@ -505,28 +511,92 @@ def __init__(
     # If the destination is a single one, we assume that we will have only one
     # job to run - and thus we avoid using temporary tables
     self.temp_tables = True if callable(destination) else False
-
+    self.kms_key = kms_key
     self._validate = validate
-    if self._validate:
-      self.verify()
 
-  def verify(self):
-    if (isinstance(self._custom_gcs_temp_location, str) and
+  def verify(self, options):
+
+    self._custom_gcs_temp_location = (
+        self._custom_gcs_temp_location
+        or options.view_as(GoogleCloudOptions).temp_location)
+
+    if (not self._custom_gcs_temp_location or
         not self._custom_gcs_temp_location.startswith('gs://')):
-      # Only fail if the custom location is provided, and it is not a GCS
-      # location.
+
+      logging.info('No appropriate location was provided to perform file loads'
+                   'to GCS.')
+      bucket = self.try_to_create_default_gcs_bucket(options)
+
+      if bucket:
+        self._custom_gcs_temp_location = 'gs://%s/temp/' % bucket.name
+        return
+
       raise ValueError('Invalid GCS location.\n'
                        'Writing to BigQuery with FILE_LOADS method requires a '
                        'GCS location to be provided to write files to be '
                        'loaded into BigQuery. Please provide a GCS bucket, or '
                        'pass method="STREAMING_INSERTS" to WriteToBigQuery.')
 
+  def try_to_create_default_gcs_bucket(self, options):
+    DEFAULT_BUCKET_NAME = "dataflow-staging-%s-%s"
+    DEFAULT_REGION = "US"
+    logging.info('Attempting to get or create a default GCS bucket.')
+
+    project_name = options.view_as(GoogleCloudOptions).project
+
+    if not project_name and isinstance(self.destination,
+                                       bigquery_api.TableReference):
+      project_name = self.destination.projectId
+
+    region = options.view_as(GoogleCloudOptions).region
+
+    if not project_name:
+      raise ValueError('--project is a required option.'
+                       ' To create a default bucket, Beam needs a project '
+                       'parameter passed to your pipeline.')
+
+    # Retrieve the project number for the default bucket
+    from google.cloud import resource_manager
+    client = resource_manager.Client()
+    project_number = client.fetch_project(project_name).number
+
+    # We get the region, and cut off the zone id if there is one.
+    region = (region or DEFAULT_REGION).split('-')[0].lower()
+
+    bucket_name = DEFAULT_BUCKET_NAME % (region, project_number)
```

Comment:
If a user wants to revoke a key that's associated to a job, and the bucket created by this method is using that key, it will affect all jobs in the same project and region. This is unintuitive behavior. Instead, `try_to_create_default_gcs_bucket` should fail if `dataflow_kms_key` is specified. (Java SDK will also change its behavior to not automatically create buckets with KMS keys.)

Reply:
FYI, I've opened https://github.com/apache/beam/pull/8830 for this issue in Java.